pkg/prober/socket.go

```diff
@@ -97,12 +124,26 @@ func (u *udpSockWrapper) Close() error {
 }
 
 func (p *Prober) initRawSocket() error {
-	rc, err := newRawSockWrapper()
-	if err != nil {
-		return fmt.Errorf("Unable to create rack socket wrapper: %v", err)
+	ipVersion := p.cfg.IPVersion
+
+	if ipVersion == 4 {
+		rc, err := newRawSockWrapper()
+		if err != nil {
+			return fmt.Errorf("Unable to create rack socket wrapper: %v", err)
+		}
+
+		p.rawConn = rc
+	}
+
+	if ipVersion == 6 {
+		rc, err := newIPv6RawSockWrapper()
+		if err != nil {
+			return fmt.Errorf("Unable to create rack socket wrapper: %v", err)
+		}
+
+		p.rawConn = rc
```

Comment:
nit: this could be more concise
```suggestion
	var rc rawSocket
	var err error
	switch p.cfg.IPVersion {
		case 4:
			rc, err = newRawSockWrapper()
		case 6:
			rc, err = newIPv6RawSockWrapper()
		default:
			return fmt.Errorf("invalid IP version: %d", p.cfg.IPVersion)
	}
	if err != nil {
		return fmt.Errorf("Unable to create raw socket wrapper: %v", err)
	}
	p.rawConn = rc
```